src/main/kotlin/org/wycliffeassociates/otter/jvm/app/widgets/resourcecard/model/ResourceCardItem.kt

```diff
@@ -1,5 +1,33 @@
 package org.wycliffeassociates.otter.jvm.app.widgets.resourcecard.model
 
-data class ResourceCardItem(
-    val title: String
-)
+import javafx.beans.property.DoubleProperty
+import javafx.beans.property.SimpleDoubleProperty
+import org.wycliffeassociates.otter.common.data.workbook.AssociatedAudio
+import org.wycliffeassociates.otter.common.data.workbook.Resource
+import org.commonmark.parser.Parser
+import org.commonmark.renderer.text.TextContentRenderer
+
+data class ResourceCardItem(val resource: Resource, val onSelect: () -> Unit) {
+    val title: String = getTitleTextContent()
+    val titleProgressProperty: DoubleProperty = resource.titleAudio.progressProperty()
+    val bodyProgressProperty: DoubleProperty? = resource.bodyAudio?.progressProperty()
+    val hasBodyAudio: Boolean = resource.bodyAudio != null
+
+    private fun AssociatedAudio.progressProperty(): DoubleProperty {
+        val progressProperty = SimpleDoubleProperty(0.0)
+        this.selected.subscribe {
+            progressProperty.set( if (it.value != null) 1.0 else 0.0)
+        }
+        return progressProperty
+    }
+
+    companion object {
+        val parser: Parser = Parser.builder().build()
+        val renderer: TextContentRenderer = TextContentRenderer.builder().build()
+    }
+
+    private fun getTitleTextContent(): String {
+        val document = parser.parse(resource.title.text)
+        return renderer.render(document)
+    }
```

Comment:
Good points. I'm actually just rendering the title as plain text though and saving html rendering for the body when it appears on the takes page.

Reply:
@sarabiaj I would like to talk to you about dagger-injectable components like this, as opposed to an object in common, like ParseMd.kt.

Reply:
I'd prefer to keep dagger to injection in the framework classes (Android Activities/Fragments, TornadoFX Views/ Fragments, maybe viewmodels)

for a component like this, I'd rather you just pass the dependency into the constructor rather than make dagger do it for you

Reply:
Sure. In this case, it seems like this can be solved without passing a dependency into the constructor. I think I can just create a MarkdownToHtml object like ParseMd that basically just holds static functions to do what I need.